app/screens/ChildSetup.tsx

```diff
@@ -181,90 +184,132 @@ const ChildSetup = ({ navigation }: Props): any => {
+                console.log("Decrypted error", error);
+                throw error;
+              });
+            const importedData = JSON.parse(await decryptedData);
+            await RNFS.writeFile(tempRealmFile, JSON.stringify(decryptedData), "utf8");
+            oldChildrenData = importedData;
+          } else {
             const exportedFileContent: any = await RNFS.readFile(decodeURIComponent(res[0].uri), 'base64');
-            const exportedFileContentRealm: any = await RNFS.writeFile(tempRealmFile, exportedFileContent, "base64");
-            let importedrealm = await new Realm({ path: 'user1.realm' });
+            await RNFS.writeFile(tempRealmFile, exportedFileContent, "base64");
+            importedrealm = await new Realm({ path: 'user1.realm' });
             if (importedrealm) {
               importedrealm.close();
             }
             importedrealm = await new Realm({ path: 'user1.realm' });
             const user1Path = importedrealm.path;
             console.log(user1Path, "..user1Path");
-            const oldChildrenData = importedrealm.objects('ChildEntity');
-            //console.log(exportedFileContentRealm, "..exportedFileContentRealm..")
-            //console.log(oldChildrenData, "..newoldChildrenData..")
-            setImportAlertVisible(false);
-            setLoading(true);
-            setIsImportRunning(true);
-            if (oldChildrenData.length > 0) {
-              await userRealmCommon.openRealm();
-              await userRealmCommon.deleteAllAtOnce();
-              setIsImportRunning(false);
-              setLoading(false);
-              navigation.navigate('ChildImportSetup', {
-                importResponse: JSON.stringify(oldChildrenData)
-              });
-              importedrealm.close();
-              try {
-                Realm.deleteFile({ path: tempRealmFile });
-              } catch (error) {
-                //console.log(error);
-              }
+            oldChildrenData = importedrealm.objects('ChildEntity');
+          }
 
+          setImportAlertVisible(false);
+          setLoading(true);
+          setIsImportRunning(true);
+          if (oldChildrenData.length > 0) {
+            await userRealmCommon.openRealm();
+            await userRealmCommon.deleteAllAtOnce();
+            setIsImportRunning(false);
+            setLoading(false);
+            navigation.navigate('ChildImportSetup', {
+              importResponse: JSON.stringify(oldChildrenData)
+            });
+            if (importedrealm) {
+              importedrealm.close();
             }
-            else {
-              setLoading(false);
-              setIsImportRunning(false);
+            try {
+              Realm.deleteFile({ path: tempRealmFile });
+            } catch (error) {
+              //console.log(error);
             }
 
           }
+          else {
+            setLoading(false);
+            setIsImportRunning(false);
+          }
```

Comment:
Let's please introduce a new method to combine lines 218-238 and 280-301

Something like:

```
function handleImportedData(importedData: any) {
  if (importedData.length > 0) {
    userRealmCommon.openRealm()
      .then(() => userRealmCommon.deleteAllAtOnce())
      .then(() => {
        setLoading(false);
        setIsImportRunning(false);
        navigation.navigate('ChildImportSetup', {
          importResponse: JSON.stringify(importedData),
        });
      })
      .catch((error) => console.error('Error handling imported data:', error));

    if (tempRealmFile.endsWith('.backup')) {
      Realm.deleteFile({ path: tempRealmFile });
    }
  } else {
    setLoading(false);
    setIsImportRunning(false);
  }
}
```